src/components/link/_link.scss

```diff
@@ -1,6 +1,11 @@
 .euiLink {
   @include euiLink;
 
+  &[target='_blank'] {
+    // Make the screen reader only text positioned relatively against the link itself
+    position: relative;
+  }
+
```

Comment:
[not a blocker, just a wild thought] For the external EuiLink use case, I'm wondering - why not skip the screen reader only text and just conditionally bake it into the `aria-label` on the external icon?

<img width="666" alt="" src="https://user-images.githubusercontent.com/549407/132747644-670a1f5d-a83b-4b22-9f06-d0c87279bdff.png">

It looks like when we implemented the SR text it was a conscious decision to allow people to disable the icon on a `target="_blank"` link, but I wonder if it wouldn't be simpler to enforce the icon+SR text on new tab/window links (the same way we enforce other semantic/accessibility best practices).

Again, definitely not a blocker and happy to merge this PR w/o, just wanted to throw that out there to see what you thought!

Reply:
This was a direct implementation from the a11y-expert-no-longer-here. I'm guessing he had his reasons.

Reply:
Oh and it can go the other way too where the icon can be present without opening in a new tab.

Reply:
Right yeah, the aria-label itself would need some conditional logic to detect `target="_blank"` and add the parenthetical, but that's fairly straightforward.

If you're not a huge fan of the concept of putting the SR text all on `aria-label` though, no worries, we can just leave it as-is 👍